### PR TITLE
add rubocop

### LINKED ISSUE #6: Set up CI for rubocop
# Problem

I added rubocop to this repository in #2, but there it has to be run manually. Since it isn't an automatic change we will be prone to introduce regressions that will make it hard to integrate with rubygems later on.

# Solution

Setup rubocop to run in github actions for every PR.
I think it would make sense to copy the configuration from rubygems, which will reduce how much we will need to change once we propose to integrate the plugin into rubygems.

[rubygems linter config](https://github.com/rubygems/rubygems/blob/master/.github/workflows/ubuntu-lint.yml)
[command to setup rubygems environment](https://github.com/rubygems/rubygems/blob/master/Rakefile#L8-L11)

### PR DESCRIPTION
**Pull request by doodzik**
fixes #6 

# Problem

I added rubocop to this repository in #2, but there it has to be run manually. Since it isn't an automatic change we will be prone to introduce regressions that will make it hard to integrate with rubygems later on.

# Solution

Setup rubocop to run in github actions for every PR.
I think it would make sense to copy the configuration from rubygems, which will reduce how much we will need to change once we propose to integrate the plugin into rubygems.

### CODE REVIEW

**Review by bettymakes — APPROVED** [review]
I haven't explored much around Github actions, but I dig your proposed solution.

Not sure if I can actually tophat the workflow locally, but with the quick skim i've done in the [intro guides](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions), your configurations look fine to me 👍 .

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,5 +25,17 @@ jobs:
...
       run: bundle exec rake
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.6
```
Comment (jchestershopify) [inline .github/workflows/main.yml:38]: Any reason for 2.6?

**Review by jchestershopify — APPROVED** [review]
I'm not a huge fan of rubocop's `{|foo| bar }` spacing inside braces, but 🤷

File: .github/workflows/main.yml (hunk 1 above)
Comment (rochlefebvre) [reply .github/workflows/main.yml:38]: Oh, which leads me to wonder: can we go lower? We may eventually work around the dependency requirements, but perhaps our code should remain compliant with older syntax? :barf:

**Review by rochlefebvre — COMMENTED** [review]
Good work!

**Review by rochlefebvre — APPROVED** [review]
👍

**Pull request merged**